match/new_orleans_pd.py

```diff
@@ -5,9 +5,61 @@
     JaroWinklerSimilarity,
     DateSimilarity,
     ColumnsIndex,
+    Swap,
 )
 from lib.post import extract_events_from_post, load_for_agency
 import pandas as pd
+from lib.clean import canonicalize_officers
+
+
+def deduplicate_pprr(pprr):
+    df = (
+        pprr.loc[
+            pprr.uid.notna(),
+            ["employee_id", "first_name", "last_name", "middle_name", "uid"],
+        ]
+        .drop_duplicates()
+        .set_index("uid", drop=True)
+    )
+
+    matcher = ThresholdMatcher(
+        ColumnsIndex("employee_id"),
+        {
+            "first_name": JaroWinklerSimilarity(),
+            "last_name": JaroWinklerSimilarity(),
+        },
+        df,
+        variator=Swap("first_name", "last_name"),
+    )
+    decision = 0.950
+    matcher.save_clusters_to_excel(
+        deba.data("match/pprr_new_orleans_pd_dedup.xlsx"), decision, decision
+    )
+    clusters = matcher.get_index_clusters_within_thresholds(decision)
+    return canonicalize_officers(pprr, clusters)
+
+
+def deduplicate_award(award):
+    df = award[["uid", "first_name", "last_name", "middle_name"]]
+    df = df.drop_duplicates(subset=["uid"]).set_index("uid")
+    df.loc[:, "fc"] = df.first_name.fillna("").map(lambda x: x[:1])
+    df.loc[:, "mc"] = df.middle_name.fillna("").map(lambda x: x[:1])
+    df.loc[:, "lc"] = df.last_name.fillna("").map(lambda x: x[:1])
+    matcher = ThresholdMatcher(
+        ColumnsIndex(["fc", "lc", "mc"]),
+        {
+            "first_name": JaroWinklerSimilarity(),
+            "last_name": JaroWinklerSimilarity(),
+        },
+        df,
+        variator=Swap("first_name", "last_name"),
+    )
```

Comment:
`fc` is the first character of `first_name`. If `fc` is in the index, only rows with the same character for first name can match. The `Swap` variator is run after the index, therefore it will not have an effect in most cases. I.e. "John Smith" will never be able to match with "Smith John" because they belong to different indices, 'j' != 's'. Same thing with `lc`. If you still want to use the `Swap` variator then please avoid using `fc` and `lc` in the index. I guess you can still use `mc`, however in a dataset that is so inconsistent that the first name and last name order can vary then how consistent can the middle name be?

Reply:
Great point. Thank you

Reply:
But then less meticulous agencies can also decide to not record the middle name, no? I'd say unless you have found instances where the name is written in different order then just avoid using the `Swap` variator. And if you do use it then please find some other columns that can be consistent. One idea I have is to sort `fc` and `lc` in alphabetical order and then concatenate them. That column will be way more resilient.

Reply:
I've removed it. This is ready for review. Thanks!